classes/Company.php

```diff
@@ -0,0 +1,224 @@
+                country = :country,
+                state = :state,
+                city = :city,
+                specialization = :specialization,
+                contact = :contact,
+                contact_person = :contact_person';
+
+            $stmt = $this->conn->prepare($query);
+
+            $this->company_name = ucwords(htmlspecialchars(strip_tags($this->company_name)));
+            $this->email = ucwords(htmlspecialchars(strip_tags($this->email)));
+            $this->country = htmlspecialchars(strip_tags($this->country));
+            $this->state = ucwords(htmlspecialchars(strip_tags($this->state)));
+            $this->city = ucwords(htmlspecialchars(strip_tags($this->city)));
+            $this->specialization = ucwords(htmlspecialchars(strip_tags($this->specialization)));
+            $this->contact = htmlspecialchars(strip_tags($this->contact));
+            $this->contact_person = htmlspecialchars(strip_tags($this->contact_person));
+
+            $stmt->bindParam(':company_name', $this->company_name);
+            $stmt->bindParam(':email', $this->email);
+            $stmt->bindParam(':country', $this->country);
+            $stmt->bindParam(':state', $this->state);
+            $stmt->bindParam(':city', $this->city);
+            $stmt->bindParam(':specialization', $this->specialization);
+            $stmt->bindParam(':contact', $this->contact);
+            $stmt->bindParam(':contact_person', $this->contact_person);
+
+            if($stmt->execute()) {
+                return true;
+            }
+
+            return false;
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while creating a company in the database! \n", 3, 'error_log'); 
+        }
+    }
+
+    public function read($limit = 10, $offset = 0) {
+        try{
+            $query = "SELECT * FROM companies ORDER BY created_at DESC LIMIT :limit OFFSET :offset";
+            $stmt = $this->conn->prepare($query);
+            $stmt->bindParam(':limit', $limit, PDO::PARAM_INT);
+            $stmt->bindParam(':offset', $offset, PDO::PARAM_INT);
+            $stmt->execute();
+            return $stmt;
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while reading a company from the database! \n", 3, 'error_log'); 
+        }
+    }
+
+    public function getTotalCompanies(){
+        try{
+            $query = 'SELECT COUNT(*) AS total FROM ' . $this->table;
+            $stmt = $this->conn->prepare($query);
+            $stmt->execute();
+            $total_candidates = $stmt->fetch(PDO::FETCH_ASSOC);
+            return $total_candidates['total'];
+        }catch(PDOException $e){
+            $timestamp = date('Y-m-d H:i:s'); 
+            error_log("$timestamp: Something went wrong while getting total number of companies from the database! \n", 3, 'error_log');
```

Comment:
Log the actual exception $e so we can see the issue in the log file for troubleshooting.

Reply:
I have done this.